src/modules/auth/auth.service.ts

```diff
@@ -106,6 +122,82 @@ export class AuthService {
     return this.generateTokens(userId);
   }
 
+  async verifyEmail(authVerifyEmailDto: AuthVerifyEmailDto): Promise<void> {
+    const { email } = authVerifyEmailDto;
+
+    const user = await this.usersService.findByEmail(email);
+
+    if (!user) {
+      throw new ConflictException('This email is not associated with any user');
+    }
+
+    if (user.isVerified) {
+      throw new ConflictException('Email is already verified');
+    }
+
+    let otp = '';
+
+    for (let i = 0; i < 6; i += 1) {
+      otp += Math.floor(Math.random() * 10);
+    }
+
+    await this.mailerService.sendMail({
+      to: user.email,
+      subject: 'Black circle OTP code email verification',
+      html: `<p>OTP code: <b>${otp}</b></p>`,
+    });
+
+    const otpToken = await this.tokenService.create<OtpCodePayloadToken>(
+      {
+        code: otp,
+        email,
+      },
+      {
+        expiresIn: '1m',
+      },
+    );
+
+    return void (await this.usersService.patch({
+      id: user.id,
+      otpToken,
+    }));
+  }
+
+  async verifyOtp(authVerifyOtp: AuthVerifyOtpDto): Promise<void> {
```

Comment:
but if you return value, it could not be void, void only if there no return value

Reply:
async function always returns promise, so if promise don't resolve any value it is equal `Promise<void>` type.
```
return void (await this.usersService.patch({
      id: user.id,
      isVerified: true,
      otpToken: null,
    }));
```
return void construction is equal to next construction
```
await this.usersService.patch({
      id: user.id,
      isVerified: true,
      otpToken: null,
    })
return undefined;
```

Reply:
why you need to return anything? this is my point. you can't write void and the make a return. if you need return find other way to describe types

Reply:
I mean it is just about syntax.
return void is same thing as no return 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/void